rfcs/BFDv2/Patient RFC.md

```diff
@@ -0,0 +1,250 @@
+	* [Future State](#future-state)
+	* [Differential View](#differential-view)
+	* [Unresolved Questions](#unresolved-questions)
+	* [Drawbacks](#proposed-solution-drawbacks)
+	* [Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Prior Art](#prior-art)
+* [Future Possibilities](#future-possibilities)
+* [Addendums](#addendums)
+
+
+## Motivation
+[Motivation]: #motivation
+
+This change will make BFD's Patient FHIR resource increase the system's conformance with FHIR R4 and industry alliance like CARIN.
+This will help BFD move its platform to use the first normative release of FHIR R4. This will also make integration and 
+interoperability with all our peer partners seamless by enabling to provide FHIR R4 validated and conformant Patient resources
+(Beneficiaries demographic data).
+
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+
+BFD proposes to provide a new v2 endpoint based on Patient FHRI-R4 and CARIN BB Profile. 
+
+Sample indicative Patient v2 endpoint:  
+https://localhost:1337/v2/fhir/Patient?_id=-19990000000001&_format=json
+
+
+Patient FHIR R4 (US-Core, Normative release):  
+http://hl7.org/fhir/R4/patient.html
+
+
+CARIN Patient Profile:  
+http://build.fhir.org/ig/HL7/carin-bb/StructureDefinition-CARIN-BB-Patient.html
+
+### Detailed Design
+
+[Detailed Design]: #detailed-design
+
+**HAPI Version to be used:** HAPI FHIR 4.1.0
+
+**HAPI 4.1.0 Patient Data Model class to be used:** org.hl7.fhir.r4.model.Patient
+
+**POC Code snippet:** 
+
+	// Slicing for Patient Identifier
+      Patient patientSlicing = new Patient();
+
+      patientSlicing.addIdentifier().setValue("-20140000010000")
+          .setSystem("https://bluebutton.cms.gov/resources/variables/bene_id").getType().addCoding()
+          .setCode(
+          "PI")
+          .setSystem("http://terminology.hl7.org/CodeSystem/v2-0203")
+          .setDisplay("Patient Internal Identifier");
+
+      patientSlicing.addIdentifier()
+          .setValue("-2b034220943953861f7b17963091ea962c13548f4b1d5f4c1013ee1779d621f4")
+          .setSystem("https://bluebutton.cms.gov/resources/identifier/mbi-hash").getType()
+          .addCoding().setCode("MC").setSystem("http://terminology.hl7.org/CodeSystem/v2-0203")
+          .setDisplay("Patient's Medicare Number");
+
+**Below are the payload changes being proposed in Patient resource:**
+
+
+#### Changes:
+
+##### Slicing:
+
+Current State: No Slicing.
+
+Future State: The element Patient.identifier is sliced based on the values of value:type, value:system.
+
+Slicing Implementation: POC code above. Slicing codes are used from the below list:
+
+Code		Display  
+MR		Medical record number  
+MA		Patient Medicaid number  
+PI		Patient internal identifier  
+PT		Patient external identifier  
+SN		Subscriber Number  
+ 
+
+##### Extensions/Code System Changes:
+
+Race:
+ 
+HL7 FHIR extension for Race used by US-Core and CARIN.
+
+##### Patient.Active field 
+
+Type boolean
```

Comment:
If you want this reviewed now, this seems incomplete: you should state how it would be derived/mapped.